pkg/liqoctl/status/pods_test.go

```diff
@@ -0,0 +1,438 @@
+// Copyright 2019-2021 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/utils/pointer"
+)
+
+var _ = Describe("Pods", func() {
+	Describe("CollectionError", func() {
+		var (
+			appType = "Deployment"
+			appName = "app"
+			err     = errors.New("ce")
+		)
+
+		Context("creating a new collectionError", func() {
+			It("should hold the passed parameters during the creation", func() {
+				ce := newCollectionError(appType, appName, err)
+				Expect(ce.appType).To(Equal(appType))
+				Expect(ce.appName).To(Equal(appName))
+				Expect(ce.err).To(MatchError(err))
+			})
+		})
+	})
+
+	Describe("PodState", func() {
+		var (
+			deploymentType = "Deployment"
+			image          = "liqo:testImage"
+			podS           podState
+		)
+		JustBeforeEach(func() {
+			podName := "pod1"
+			err := errors.New("pod1")
+			podS = podState{
+				errorFlag:      false,
+				deploymentType: "",
+				desired:        1,
+				ready:          1,
+				available:      1,
+				unavailable:    0,
+				imageVersions:  []string{image},
+				errors:         errorCountMap{podName: &errorCount{[]error{err}}},
+			}
+		})
+
+		Describe("creating a new podState", func() {
+			It("should hold the passed parameters during the creation", func() {
+				ps := newPodState(deploymentType)
+				Expect(ps.deploymentType).To(Equal(deploymentType))
+			})
+		})
+
+		Describe("handling images field", func() {
+			Context("getting images", func() {
+				It("should return a slice of length 1", func() {
+					im := podS.getImages()
+					Expect(len(im)).To(BeNumerically("==", 1))
```

Comment:
```suggestion
					Expect(im).To(HaveLen(1))
```

Nit. Just a bit more idiomatic